packages/medusa/src/interfaces/payment-processor.ts

```diff
@@ -0,0 +1,219 @@
+import {
+  Address,
+  Customer,
+  PaymentSessionStatus,
+  ShippingMethod,
+} from "../models"
+import { MedusaContainer } from "../types/global"
+
+export type PaymentProcessorContext = {
```

Comment:
question(non-blocking): Would it make sense to include a `context` or `details` object to store information specific to the provider being used?

Reply:
what information are you thinking of?

Reply:
An example is Adyen which in some instances need the customer IP to perform fraud checks; having an open ended prop that can take processor specific data like this can be helpful.

Reply:
I will add it indeed, how is it managed at the moment?